diff --git a/mgrs.js b/mgrs.js
index ba902e159cc8c463cf1c6bc0a8044747f41737f3..896e5291a328e86e62b22a9966fb1eb9f5b3a38b 100644
--- a/mgrs.js
+++ b/mgrs.js
@@ -209,9 +209,9 @@ class Mgrs {
 
         const { zone, band, e100k, n100k, easting, northing } = this;
 
-        // truncate to required precision
-        const eRounded = Math.floor(easting/Math.pow(10, 5-digits/2));
-        const nRounded = Math.floor(northing/Math.pow(10, 5-digits/2));
+        // round to required precision (Math.round instead of Math.floor to handle floating-point precision errors)
+        const eRounded = Math.round(easting/Math.pow(10, 5-digits/2));
+        const nRounded = Math.round(northing/Math.pow(10, 5-digits/2));
 
         // ensure leading zeros
         const zPadded = zone.toString().padStart(2, '0');
