diff --git a/mgrs.js b/mgrs.js
index ba902e159cc8c463cf1c6bc0a8044747f41737f3..07145e72de6a2e415f304b9145e86a811d8019f7 100644
--- a/mgrs.js
+++ b/mgrs.js
@@ -250,7 +250,7 @@ class Utm_Mgrs extends Utm {
         // convert UTM to lat/long to get latitude to determine band
         const latlong = this.toLatLon();
         // grid zones are 8° tall, 0°N is 10th band
-        const band = latBands.charAt(Math.floor(latlong.lat/8+10)); // latitude band
+        const band = latBands.charAt(Math.floor(latlong.lat.toFixed(12)/8+10)); // latitude band
 
         // columns in zone 1 are A-H, zone 2 J-R, zone 3 S-Z, then repeating every 3rd zone
         const col = Math.floor(this.easting / 100e3);
diff --git a/utm.js b/utm.js
index 281b7090b383ccd02ad30a8740450d650138280c..80e2d8b142afffe2828afe5989bceb09b7a1b090 100644
--- a/utm.js
+++ b/utm.js
@@ -60,8 +60,8 @@ class Utm {
         if (typeof hemisphere != 'string' || !hemisphere.match(/[NS]/i)) throw new RangeError(`invalid UTM hemisphere ‘${hemisphere}’`);
         if (verifyEN) { // range-check E/N values
             if (!(0<=easting && easting<=1000e3)) throw new RangeError(`invalid UTM easting ‘${easting}’`);
-            if (hemisphere.toUpperCase()=='N' && !(0<=northing && northing<9328094)) throw new RangeError(`invalid UTM northing ‘${northing}’`);
-            if (hemisphere.toUpperCase()=='S' && !(1118414<northing && northing<=10000e3)) throw new RangeError(`invalid UTM northing ‘${northing}’`);
+            if (hemisphere.toUpperCase()=='N' && !(0<=northing && northing<9329006)) throw new RangeError(`invalid UTM northing ‘${northing}’`);
+            if (hemisphere.toUpperCase()=='S' && !(1116914<northing && northing<=10000e3)) throw new RangeError(`invalid UTM northing ‘${northing}’`);
         }
         if (!datum || datum.ellipsoid==undefined) throw new TypeError(`unrecognised datum ‘${datum}’`);
 
